numpy/linalg/linalg.py

```diff
@@ -1334,48 +1343,77 @@ def svd(a, full_matrices=True, compute_uv=True):
     Notes
     -----
 
-    .. versionadded:: 1.8.0
-
-    Broadcasting rules apply, see the `numpy.linalg` documentation for
-    details.
-
-    The decomposition is performed using LAPACK routine _gesdd
+    The decomposition is performed using LAPACK routine ``_gesdd``.
 
-    The SVD is commonly written as ``a = U S V.H``.  The `v` returned
-    by this function is ``V.H`` and ``u = U``.
+    SVD is usually described for the factorization of a 2D matrix :math:`A`.
+    The higher-dimensional case will be discussed below. In the 2D case, SVD is
+    written as :math:`A = U S V^H`, where :math:`A=` `a`, :math:`U=` `u`,
+    :math:`S=` ``np.diag(s)`` and :math:`V^H=` `vh`. The 1D array `s` contains
+    the singular values of `a` and `u` and `vh` are unitary. The rows of `vh`
+    are the eigenvectors of :math:`A^H A` and the columns of `u` are the
+    eigenvectors of :math:`A A^H`. In both cases the corresponding (possibly
+    non-zero) eigenvalues are given by ``s**2``.
 
-    If ``U`` is a unitary matrix, it means that it
-    satisfies ``U.H = inv(U)``.
+    If `a` is a ``matrix`` object (as opposed to an ``ndarray``), then so are all
+    the return values.
 
-    The rows of `v` are the eigenvectors of ``a.H a``. The columns
-    of `u` are the eigenvectors of ``a a.H``.  For row ``i`` in
-    `v` and column ``i`` in `u`, the corresponding eigenvalue is
-    ``s[i]**2``.
+    .. versionadded:: 1.8.0
```

Comment:
I'd move the `.. versionadded::` up to the top of the notes with a one line explanation, see other functions where that is done. The expanded explanation can be left here, maybe beginning `If a has more than two dimensions, then broadcasting rules apply as explained in ....`

Reply:
sounds good too.

Reply:
One more question. I've seen two styles to do `.. versionadded` in linalg.py:

```
    .. versionchanged:: 1.14
       Can now operate on stacks of matrices
```

and

```
    .. versionadded:: 1.8.0

    Broadcasting rules apply, see the `numpy.linalg` documentation for
    details.
```

The former seems clearer to me but is less common. It renders approximately as:

Changed in version 1.14: Can now operate on stacks of matrices

I.e. everything on one line, so it is clear that following paragraphs are not related to changes in a given version. Would you agree this is better?

Reply:
Hard to pick between the two, but the first does seem a hair better. Note that the indented message also works for `.. versionadded::`

Reply:
Now I get it. Just to be sure, I checked also the documentation of the `os` module of Python, which contains plenty of examples and I also checked the Sphinx docs to make sure I get it right.

A lot of docstrings in linalg.py, which mostly use `versionadded` where it should in most cases be `versionchanged`, are missing the indentation. This confused when I was looking at other examples. I'd rather do it right here.